sway.5.txt

```diff
@@ -175,3 +179,34 @@ Commands
 **workspace** <name> output <output>::
 	Specifies that the workspace named _name_ should appear on the specified
 	_output_.
+
+Criteria
+--------
+
+A criteria is a string in the form of e.g.:
+
+	[class="[Rr]egex.*" title="some title"]
+
+The string contains one or more (space separated) attribute/value pairs and they
+are used by some commands filter which views to execute actions on. All attributes
+must match for the criteria string to match.
+
+Currently supported attributes:
+
+**class**::
+	Compare value against the window class. Can be a regular expression. If value
+	is _focused_ then the window class must be the same as that of the currently
+	focused window.
+
+**id**::
+	Compare against the app id. Must be a normal string.
```

Comment:
```
[handlers.c:136] handle:11 type:0 state:0 parent:0 mask:1 (x:0 y:0 w:722 h:391) title:Terminal class:(null) appid:gnome-terminal-server
[handlers.c:136] handle:12 type:0 state:0 parent:0 mask:1 (x:0 y:0 w:1920 h:1059) title:Volume Control class:(null) appid:pavucontrol
[handlers.c:136] handle:8 type:8 state:0 parent:0 mask:1 (x:0 y:0 w:112 h:818) title:gkrellm class:gkrellm appid:(null)
```
Ah, turns out in wayland (or specifically xdg-shell) app_id is a string, e.g. dbus service name.

https://github.com/Cloudef/wlc/blob/23cbe58d958911836698841aa1aab1c58f6c453d/protos/xdg-shell.xml#L197-L221